ultralytics/engine/trainer.py

```diff
@@ -762,3 +768,24 @@ def build_optimizer(self, model, name="auto", lr=0.001, momentum=0.9, decay=1e-5
             f'{len(g[1])} weight(decay=0.0), {len(g[0])} weight(decay={decay}), {len(g[2])} bias(decay=0.0)'
         )
         return optimizer
+
+    def _setup_hub(self):
+        """Setup session for Hub Training."""
+        from ultralytics.utils import SETTINGS
+
+        if SETTINGS["hub"] is False or self.hub_session is not None:
```

Comment:
```diff
- if SETTINGS["hub"] is False or self.hub_session is not None:
+ if SETTINGS["hub"] is False or self.hub_session is None:
```

Reply:
@Burhan-Q I think the case here is to return directly if `if SETTINGS["hub"]=False` or `hub_session` has already been created no?

Reply:
Yes, you're right. At first glance, I was thinking that the `self.hub_session` would be `None` in most cases since it's set during `init` but I later noticed that the `model.session` is applied to `trainer.hub_session` post-init